modules/ximgproc/include/opencv2/ximgproc/edge_drawing.hpp

```diff
@@ -0,0 +1,81 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef __OPENCV_EDGE_DRAWING_HPP__
+#define __OPENCV_EDGE_DRAWING_HPP__
+
+#include <opencv2/core.hpp>
+
+namespace cv
+{
+namespace ximgproc
+{
+
+//! @addtogroup ximgproc_fast_line_detector
+//! @{
+
+class CV_EXPORTS_W EdgeDrawing : public Algorithm
+{
+public:
+
+    enum GradientOperator
+    {
+        PREWITT = 0,
+        SOBEL = 1,
+        SCHARR = 2,
+        LSD = 3
+    };
+
+    enum ImageStyle
+    {
+        DRAW_NONE = 0,
+        DRAW_CIRCLES = 1,
+        DRAW_ELLIPSES = 2,
+        DRAW_BOTH = 3
+    };
+
+    enum EllipseFittingMethods
+    {
+        BOOKSTEIN = 0,
+        FPF = 1
+    };
+
+    CV_WRAP virtual void setInput(InputArray image) = 0;
+    CV_WRAP virtual void getEdgeImage(OutputArray dst) = 0;
+    CV_WRAP virtual void getGradImage(OutputArray dst) = 0;
+
+    CV_WRAP virtual std::vector<std::vector<Point> > getSegments() = 0;
+
+    CV_WRAP virtual void getLines(InputArray src, OutputArray lines, bool validate = true) = 0;
+    CV_WRAP virtual void detectEllipses(OutputArray ellipses, bool validate = true) = 0;
+    CV_WRAP virtual void drawEllipses(InputOutputArray colorImage, int style) = 0;
+
+    virtual ~EdgeDrawing() { }
+
+protected:
+    int width; // width of source image
+    int height; // height of source image
+    uchar* srcImg;
+    std::vector<std::vector<Point> > segmentPoints;
+    double sigma; // Gaussian sigma
+    Mat smoothImage;
+    uchar* edgeImg; // pointer to edge image data
+    uchar* smoothImg; // pointer to smoothed image data
+    int segmentNos;
+    int minPathLen;
+    Mat srcImage;
+
+};
+
+
+CV_EXPORTS_W Ptr<EdgeDrawing> createEdgeDrawing(int op = EdgeDrawing::SOBEL,
+    int _gradThresh = 20, int _anchorThresh = 0, int _scanInterval = 1,
+    int _minPathLen = 10, double _sigma = 1.0, bool _sumFlag = true);
```

Comment:
We should avoid underscore prefixes in args of function declarations (they go into documentation and bindings, like Python)